internal/model/model.go

```diff
@@ -1706,6 +1706,8 @@ func (m *Model) CheckFolderHealth(id string) error {
 			err = errors.New("folder path missing")
 		} else if !folder.HasMarker() {
 			err = errors.New("folder marker missing")
+		} else if space, syserr := osutil.FreeDiskSpaceMB(folder.Path()); syserr != nil && space < uint64(folder.MinimumDiskSpaceMB) {
```

Comment:
We might still fill the disk if we're trying to pull a file that's larger than 500 MB.